src/model.cpp

```diff
@@ -475,6 +475,45 @@ bool Model::hasImports() const
     return importsPresent;
 }
 
+void fixComponentUnits(const ModelPtr &model, const ComponentPtr &component)
+{
+    for (size_t v = 0; v < component->variableCount(); ++v) {
+        auto variable = component->variable(v);
+        if (variable->units() != nullptr) {
+            // Find the units in the model and switch out.
+            auto units = model->units(variable->units()->name());
+            if (units != nullptr) {
+                variable->setUnits(units);
+            }
+        }
+    }
+    for (size_t c = 0; c < component->componentCount(); ++c) {
+        fixComponentUnits(model, component->component(c));
+    }
+}
+
+void fixImportSourceUnits(const ImportSourcePtr &i1, ImportSourcePtr &i2)
```

Comment:
Why `ImportSourcePtr &i2` and not `const ImportSourcePtr &i2`?

Reply:
Because it's changing.

Reply:
Where? As far as I can tell, it's used to retrieve its owning model and set the import source of `u2`, nothing else.

Reply:
in setting the import source of u2, the import source itself is updated.  Two way street ... see the setImportSource functions in the Component and Units classes.

Reply:
Was being lazy, so thanks for pointing me in the right direction.